components/gfx/display_list/mod.rs

```diff
@@ -1038,3 +1135,17 @@ impl fmt::Debug for DisplayItem {
     }
 }
 
+impl SizeOf for DisplayItem {
+    fn size_of_excluding_self(&self) -> usize {
+        match *self {
+            SolidColorClass(ref item) => item.size_of_excluding_self(),
+            TextClass(ref item)       => item.size_of_excluding_self(),
+            ImageClass(ref item)      => item.size_of_excluding_self(),
+            BorderClass(ref item)     => item.size_of_excluding_self(),
+            GradientClass(ref item)   => item.size_of_excluding_self(),
+            LineClass(ref item)       => item.size_of_excluding_self(),
+            BoxShadowClass(ref item)  => item.size_of_excluding_self(),
+        }
+    }
+}
```

Comment:
Eventually with custom compiler macros it may well be nice to have this autogenerated, but obviously you don't have to do that right now.